resources/fetchers/ecr_fetcher.go

```diff
@@ -34,8 +35,15 @@ import (
 )
 
 const (
-	PrivateRepoRegexTemplate = "^%s\\.dkr\\.ecr\\.%s\\.amazonaws\\.com\\/([-\\w\\.\\/]+)[:,@]?"
+	// PrivateRepoRegexTemplate should identify images with an ecr regex template
+	// <account-id>.dkr.ecr.<region>.amazonaws.com/<repository-name>
+	PrivateRepoRegexTemplate = "^%s\\.dkr\\.ecr\\.([-\\w]+)\\.amazonaws\\.com\\/([-\\w\\.\\/]+)[:,@]?"
+	// PublicRepoRegex should identify images with a public ecr regex template
+	// public.ecr.aws/<aws-alias>/<repository>
 	PublicRepoRegex          = "public\\.ecr\\.aws\\/\\w+\\/([-\\w\\.\\/]+)\\:?"
```

Comment:
if this is the expected format, using regex to extract the repo seems like overkill, right?

Reply:
I am using the regex for two things:
1. Validating the format of the image so I can be sure it's an ECR image.
2. Extract the repository name out of the format.

Do you have another suggestion?